### DISCUSSION TITLE
giscus integration

### DISCUSSION THREAD

**Discussion by nicholaswilde** [squidfunk/mkdocs-material#4435] [Q&A] [answered]
## 🥅 Goal

I'm trying to enable [giscus comments](https://giscus.app/) on my [recipes site](https://nicholaswilde.io/recipes/). I've add the comments to [this page](https://nicholaswilde.io/recipes/asian/coconut-curried-vegetables-with-rice/) only for now as a test.

## 🤔 Problem

The giscus commenting system does not seem to show up at the bottom of the recipes page.

## ❓ Question

What else do I need to do to enable the giscus? 

I've only enabled the `overrides/partials/comments.html` partial but it doesn't seem to take.  Looking at the [generated html page](https://raw.githubusercontent.com/nicholaswilde/recipes/5ffa9c39a50db858f08e54cb0d981cf270dec9a0/asian/coconut-curried-vegetables-with-rice/index.html) of `coconut-curried-vegetables-with-rice`doesn't have giscus in it.

The only other examples I could find seemed to not use the `comments.html` partial like the instructions say to do.

## 👣 Steps performed

I've followed [these instructions: Adding a comment system](https://squidfunk.github.io/mkdocs-material/setup/adding-a-comment-system/). 

1. Installed the [Giscus GitHub App](https://github.com/apps/giscus)
2. Enabled [Discussions on nicholaswilde/recipes repo](https://github.com/nicholaswilde/recipes/discussions).
3. Generated snippet. Source: 
```html
 <script src="https://giscus.app/client.js"
    data-repo="nicholaswilde/recipes"
    data-repo-id="R_kgDOHIKkPg"
    data-category="Comments"
    data-category-id="DIC_kwDOHIKkPs4CRWOL"
    data-mapping="pathname"
    data-strict="0"
    data-reactions-enabled="1"
    data-emit-metadata="1"
    data-input-position="bottom"
    data-theme="light"
    data-lang="en"
    crossorigin="anonymous"
    async>
  </script>
```
4. Added `comments: true` front matter to the page. [Source](https://github.com/nicholaswilde/recipes/blob/4280e1a421a76c4f8692579fd68fe50fd8ab8831/docs/asian/coconut-curried-vegetables-with-rice.md).
```markdown
---
comments: true
---
# :coconut: Coconut Curried Vegetables with Rice
...
```
5. Created `overrides/partials/comments.html` override. [Source](https://github.com/nicholaswilde/recipes/blob/4280e1a421a76c4f8692579fd68fe50fd8ab8831/overrides/partials/comments.html).
```html
{% if page.meta.comments %}
  <h2 id="__comments">{{ lang.t("meta.comments") }}</h2>
  <!-- Insert generated code here -->
  <script src="https://giscus.app/client.js"
    data-repo="nicholaswilde/recipes"
    data-repo-id="R_kgDOHIKkPg"
    data-category="Comments"
    data-category-id="DIC_kwDOHIKkPs4CRWOL"
    data-mapping="pathname"
    data-strict="0"
    data-reactions-enabled="1"
    data-emit-metadata="1"
    data-input-position="bottom"
    data-theme="light"
    data-lang="en"
    crossorigin="anonymous"
    async>
  </script>
  <!-- Synchronize Giscus theme with palette -->
  <script>
    var giscus = document.querySelector("script[src*=giscus]")

    /* Set palette on initial load */
    var palette = __md_get("__palette")
    if (palette && typeof palette.color === "object") {
      var theme = palette.color.scheme === "slate" ? "dark" : "light"
      giscus.setAttribute("data-theme", theme)
    }

    /* Register event handlers after documented loaded */
    document.addEventListener("DOMContentLoaded", function() {
      var ref = document.querySelector("[data-md-component=palette]")
      ref.addEventListener("change", function() {
        var palette = __md_get("__palette")
        if (palette && typeof palette.color === "object") {
          var theme = palette.color.scheme === "slate" ? "dark" : "light"

          /* Instruct Giscus to change theme */
          var frame = document.querySelector(".giscus-frame")
          frame.contentWindow.postMessage(
            { giscus: { setConfig: { theme } } },
            "https://giscus.app"
          )
        }
      })
    })
  </script>
{% endif %}
```
6. Added `overrides` dir to `mkdocs.yml`. [Source](https://github.com/nicholaswilde/recipes/blob/4280e1a421a76c4f8692579fd68fe50fd8ab8831/mkdocs.yml#L18).
```diff
---
site_name: Recipes
site_description: A collection of my recipes.
site_author: Nicholas Wilde
site_url: https://nicholaswilde.io/recipes
copyright: Copyright &copy; 2022 Nicholas Wilde
dev_addr: '0.0.0.0:8000'

repo_url: https://github.com/nicholaswilde/recipes
repo_name: nicholaswilde/recipes
edit_uri: "edit/main/docs/"

theme:
  name: material
  icon:
    repo: fontawesome/brands/github
    logo: fontawesome/solid/bowl-rice
+  custom_dir: overrides
...
```

## 📁 Directory structure
```
recipes
├─ overrides
│  └─ partials
│     └─ comments.html
└─ mkdocs.yml
```

**Comment by squidfunk** [comment, marked as answer]
Which version of Material for MkDocs are you using? The `comments` partial was introduced in [8.4.4](https://github.com/squidfunk/mkdocs-material/releases/tag/8.4.4). Make sure that you are at least on that version. Other than that your configuration looks good.

**Comment by nicholaswilde** [comment]
> Which version of Material for MkDocs are you using? The `comments` partial was introduced in [8.4.4](https://github.com/squidfunk/mkdocs-material/releases/tag/8.4.4). Make sure that you are at least on that version.

I am using [mhausenblas/mkdocs-deploy-gh-pages](https://github.com/mhausenblas/mkdocs-deploy-gh-pages) GH Action which looks like is using [8.4.1](https://github.com/mhausenblas/mkdocs-deploy-gh-pages/blob/19d5a44e72e4d7cfc7792eb67b95c15edb64d774/Dockerfile#L1). That might be the issue. I will give an update once I have tested it.

Thanks 🙏🏽 for the quick reply.

**Comment by nicholaswilde** [comment]
Yep, that was it. I changed [my GH Action](https://github.com/nicholaswilde/recipes/blob/2a4c786873623268ddfa9ccacb7fa86cb8d92445/.github/workflows/ci.yaml) to the [suggested one in the docs](https://squidfunk.github.io/mkdocs-material/publishing-your-site/#with-github-actions) and it worked.

**Reply by squidfunk** [reply]
Great! Using [our recommended workflow](https://squidfunk.github.io/mkdocs-material/publishing-your-site/#with-github-actions) makes sure you're always using the latest version.

**Comment by krahets** [comment]
Thanks! It seems mkdocs-material can not be upgraded by `pip` automatically, so I reinstalled it. Now giscus works well!
